curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -0,0 +1,382 @@
+                {
+                    path = event.getName();
+                }
+                else if ( event.getResultCode() == KeeperException.Code.NOAUTH.intValue() )
+                {
+                    log.warn("Client does not have authorisation to write ephemeral node at path {}", event.getPath());
+                    authFailure.set(true);
+                    return;
+                }
+                if ( path != null )
+                {
+                    authFailure.set(false);
+                    nodePath.set(path);
+                    watchNode();
+
+                    if ( nodeExists )
+                    {
+                        client.setData().inBackground(setDataCallback).forPath(getActualPath(), getData());
+                    }
+                    else
+                    {
+                        initialisationComplete();
+                    }
+                }
+                else
+                {
+                    createNode();
+                }
+            }
+        };
+
+        createMethod = useProtection ? client.create().creatingParentContainersIfNeeded().withProtection() : client.create().creatingParentContainersIfNeeded();
+        this.data.set(Arrays.copyOf(data, data.length));
+    }
+
+    private void initialisationComplete()
+    {
+        CountDownLatch localLatch = initialCreateLatch.getAndSet(null);
+        if ( localLatch != null )
+        {
+            localLatch.countDown();
+        }
+    }
+
+    /**
+     * You must call start() to initiate the persistent ephemeral node. An attempt to create the node
+     * in the background will be started
+     */
+    public void start()
+    {
+        Preconditions.checkState(state.compareAndSet(State.LATENT, State.STARTED), "Already started");
+
+        client.getConnectionStateListenable().addListener(connectionStateListener);
+        createNode();
+    }
+
+    /**
+     * Block until the either initial node creation initiated by {@link #start()} succeeds or
+     * the timeout elapses.
+     *
+     * @param timeout the maximum time to wait
+     * @param unit    time unit
+     * @return if the node was created before timeout
+     * @throws InterruptedException if the thread is interrupted
+     */
+    public boolean waitForInitialCreate(long timeout, TimeUnit unit) throws InterruptedException
+    {
+        Preconditions.checkState(state.get() == State.STARTED, "Not started");
+
+        CountDownLatch localLatch = initialCreateLatch.get();
+        return (localLatch == null) || localLatch.await(timeout, unit);
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        if ( !state.compareAndSet(State.STARTED, State.CLOSED) )
+        {
+            return;
+        }
+
+        client.getConnectionStateListenable().removeListener(connectionStateListener);
+
+        try
+        {
+            deleteNode();
```

Comment:
I think it makes sense still. The idea is that the node persists until the instance is closed.